### ISSUE TITLE
Calculating diffraction pattern above approx. 150 degrees in 2theta

### ISSUE THREAD

**Issue by AndrewSazonov** [EasyScience/PyCrysFML#20]
Attempting to calculate the powder diffraction pattern for 2theta > 150deg results in non-zero intensities for only about half of the 2theta range. This can be seen at any of the following links:

  * [Python script](https://github.com/EasyScience/PyCrysFML/blob/issues/issues/pyCFML/cfml_utilities/powder_pattern_from_json/above_150deg/above_150deg.py)
  * [Jupyter notebook (GitHub, non-interactive)](https://github.com/EasyScience/PyCrysFML/blob/issues/issues/pyCFML/cfml_utilities/powder_pattern_from_json/above_150deg/above_150deg.ipynb)
  * [Jupyter notebook (nbviewer, non-interactive)](https://nbviewer.jupyter.org/github/EasyScience/PyCrysFML/blob/issues/issues/pyCFML/cfml_utilities/powder_pattern_from_json/above_150deg/above_150deg.ipynb)
  * [Jupyter notebook (Binder, interactive)](https://mybinder.org/v2/gh/EasyScience/PyCrysFML/issues?filepath=issues/pyCFML/cfml_utilities/powder_pattern_from_json/above_150deg/above_150deg.ipynb)

![above_150deg](https://github.com/user-attachments/assets/1446b0a1-9caa-4f98-9c5a-552dd58c0245)

**Comment by AndrewSazonov** [comment]
Fixed in CrysFML08 https://code.ill.eu/scientific-software/CrysFML2008/-/commit/e412309ce057d447cb2f99dfad7dc78b94d38b7e

![Figure_1](https://github.com/user-attachments/assets/e46d240d-936f-4906-a921-fd54ebceeb18)

**Comment by AndrewSazonov** [comment]
I am reopening this issue because, although the original problem seems to be fixed (as shown in the screenshot above), a similar issue has been observed for another system.

Specifically:
- When generating a diffraction pattern with a maximum of 160°, peaks at 120° and 140° are visible.
- When using a maximum of 150°, the peak at 140° disappears, while the one at 120° remains.
- When using a maximum of 140°, the peak at 120° also disappears.

### Reproducible example

A minimal Python script to reproduce the issue:
https://raw.githubusercontent.com/easyscience/deps-pycrysfml/refs/heads/master/issues/pyCFML/cfml_utilities/powder_pattern_from_json/above_150deg/above_150deg_new.py

### Illustration

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/a6382efe-0527-486d-83a8-59b728d1ffb4" />

**Comment by nebil77** [comment]
There was a bug in the subroutine Get_MaxNumRef of the module CFML_Reflections of CrysFML. The number of total reflections was not well estimated when the lattice parameters were small. After fixing the bug, the calculation runs well.

<img width="580" height="447" alt="Image" src="https://github.com/user-attachments/assets/6035df0d-78d0-41ea-95a2-f2090f6fae2f" />

**Comment by AndrewSazonov** [comment]
It works now when the x-data is passed to pycrysfml as min, max, and increment values, for example:

```json
"_pd_meas_2theta_range_min": 10.0,
"_pd_meas_2theta_range_max": 155.45,
"_pd_meas_2theta_range_inc": 0.05
```

However, the high-angle peaks are still truncated when the x-data is passed as an explicit array (as implemented in #31), for example:

```json
"_pd_meas_2theta_scan": [10.0, 10.05, 10.1, 10.15, 10.2, ..., 155.45]
```

Here is a Python script that reproduces the issue:
https://github.com/easyscience/deps-pycrysfml/tree/crysfml-fix/issues/pyCFML/cfml_utilities/powder_pattern_from_json/custom_x

Illustration (blue peak at about 151 deg.):
<img width="1050" height="480" alt="Image" src="https://github.com/user-attachments/assets/b36be058-2505-43d0-90b8-d6ed30a6a49c" />

**Comment by nebil77** [comment]
For x-data provided by the user, 2theta max was not computed from the x-data, but the default value of 150 was still used. This bug has been fixed and now the calculations runs well.

<img width="611" height="466" alt="Image" src="https://github.com/user-attachments/assets/7ae78efe-ba3d-4851-98d8-94a29568d5de" />
